src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -157,11 +160,23 @@ public void doMove(final Map<Territory, ProTerritory> attackMap, final IMoveDele
     ProMoveUtils.calculateAmphibRoutes(player, moveUnits, moveRoutes, transportsToLoad, attackMap, true);
     ProMoveUtils.doMove(moveUnits, moveRoutes, transportsToLoad, moveDel);
 
-    // Calculate attack routes and perform moves
+    // Calculate bombard routes and perform moves
     moveUnits.clear();
     moveRoutes.clear();
     ProMoveUtils.calculateBombardMoveRoutes(player, moveUnits, moveRoutes, attackMap);
     ProMoveUtils.doMove(moveUnits, moveRoutes, null, moveDel);
+
+    // Calculate bombing routes and perform moves
+    moveUnits.clear();
+    moveRoutes.clear();
+    isBombing = true;
```

Comment:
Yep its actually pretty bad but this is the best way I could think of to do it. The reason for this is when you move a bomber to a territory that can be bombed you get a popup (as a human) asking whether you want to bomb or attack. The same thing happens for the AI so as it calling the delegate to move bombers, a call back from the engine occurs where the AI needs to return bombing=true.